### PR TITLE
RavenDB-16651 OLAP tables + sample scripts (RavenDB-16651, RavenDB-16…

### PR DESCRIPTION
**Pull request by Danielle9897**
…650, RavenDB-16680)

### CODE REVIEW

File: src/Raven.Studio/wwwroot/App/views/database/tasks/transformationScriptSyntax.html [hunk 1]
```diff
@@ -3,27 +3,40 @@
...
            </div>
            <div data-bind="if: etlType() === 'Olap'" class="copy-area">
-               <h3 class="margin-bottom margin-bottom-lg margin-top">Simple OLAP ETL Transformation Script</h3>
+               <h3 class="margin-bottom margin-bottom-lg margin-top">Sample OLAP ETL Transformation Scripts</h3>
+               <div>Transform with Partitions</div>
                <div class="margin-bottom">
-                   <pre class="copy-area"><a href="#" class="copy-button pull-right" title="Copy code sample" data-bind="click: copySample"><i class="icon-copy"></i></a><span data-bind="html: olapEtlSampleHtml"></span></pre>
+                   <pre class="copy-area"><a href="#" class="copy-button pull-right" title="Copy code sample" data-bind="click: _.partial($root.copySample, 'olapEtlSamplePartition')"><i class="icon-copy"></i></a><span data-bind="html: olapEtlSamplePartitionHtml"></span></pre>
+               </div>
+               <div>Transfrom without Partitions</div>
```
Comment (ml054) [inline src/Raven.Studio/wwwroot/App/views/database/tasks/transformationScriptSyntax.html]: Transform

File: src/Raven.Studio/typescript/viewmodels/database/tasks/transformationScriptSyntax.ts [hunk 2]
```diff
@@ -117,22 +130,56 @@ loadToOrders(orderData);`;
...
+loadToOrders(noPartition(), {
+    Company : this.Company
+});`;
+
+    olapEtlSampleNoPartitionHtml = ko.pureComputed(() => {
+        return Prism.highlight(transformationScriptSyntax.olapEtlSampleNoPartitionText, (Prism.languages as any).javascript);
+    });
+
+    static readonly olapEtlSampleKeyText = 
+`var key = new Date(this.OrderedAt);
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/database/tasks/transformationScriptSyntax.ts:156]: we do want comment that date has special meaning here? (IIRC key will be dt?)

File: src/Raven.Studio/wwwroot/App/views/database/tasks/editOlapEtlTask.html [hunk 3]
```diff
@@ -219,6 +169,55 @@ <h3 class="margin-top margin-top-sm">No Olap Tables have been defined</h3>
...
+                </div>
+                <div data-bind="foreach: editedOlapEtl().olapTables">
+                    <div class="panel panel-hover">
+                        <div class="padding padding-sm flex-horizontal item">
+                            <div class="row flex-grow">
+                                <div class="col-sm-4">
+                                    <div class="widthlabel"><label class="superscript">Table</label><strong data-bind="text: tableName"></strong><span class="text-warning" data-bind="visible: dirtyFlag().isDirty">&nbsp;*</span></div>
+                                </div>
+                                <div class="col-sm-4">
+                                    <div class="widthlabel"><label class="superscript">Id Column Name</label><strong data-bind="text: documentIdColumn"></strong></div>
```
Comment (ml054) [inline src/Raven.Studio/wwwroot/App/views/database/tasks/editOlapEtlTask.html]: sometimes you use ID vs Id. Let's unify.

File: src/Raven.Studio/typescript/viewmodels/database/tasks/transformationScriptSyntax.ts [hunk 4]
```diff
@@ -117,22 +130,56 @@ loadToOrders(orderData);`;
...
+loadToOrders(partitionBy(key), {
+    Company : this.Company
+});`;
+
+    olapEtlSampleKeyHtml = ko.pureComputed(() => {
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/database/tasks/transformationScriptSyntax.ts]: olapEtlSampleKeyHtml = transformationScriptSyntax.highlightJavascript(this.olapEtlSampleKeyText)
	
	
	
	static highlightJavascript(source: string) {
	  return Prism.highlight(source, (Prism.languages as any).javascript);
	}

File: src/Raven.Studio/wwwroot/App/views/database/tasks/transformationScriptSyntax.html (hunk 1 above)
Comment (Danielle9897) [reply src/Raven.Studio/wwwroot/App/views/database/tasks/transformationScriptSyntax.html]: done

File: src/Raven.Studio/wwwroot/App/views/database/tasks/editOlapEtlTask.html (hunk 3 above)
Comment (Danielle9897) [reply src/Raven.Studio/wwwroot/App/views/database/tasks/editOlapEtlTask.html]: done

File: src/Raven.Studio/typescript/viewmodels/database/tasks/transformationScriptSyntax.ts (hunk 2 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/database/tasks/transformationScriptSyntax.ts:156]: adding:
`// The partition that will be created will be: "_partition={key}"`

File: src/Raven.Studio/typescript/viewmodels/database/tasks/transformationScriptSyntax.ts (hunk 4 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/database/tasks/transformationScriptSyntax.ts]: done

File: src/Raven.Studio/typescript/viewmodels/database/tasks/transformationScriptSyntax.ts [hunk 5]
```diff
@@ -113,26 +126,55 @@ orderData.TotalCost = Math.round(orderData.TotalCost * 100) / 100;
...
-// The order of the following array values determines the partitions order in parquet file path
-loadToOrders(partitionBy([['year', year], ['month', month], ['source', $custom_field]]),
-{
+loadToOrders(partitionBy(['year', year], ['month', month]), {
+    // The order of params in the partitionBy method determines the parquet file path
     Company : this.Company,
     ShipVia : this.ShipVia
+    // Note: 2 more field are always created per table by default:
+    //       * _id: The ID column - can be overriden in the task definition
+    //       * _lastModifiedTicks: The document's last modification time column - cannot be overriden
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/database/tasks/transformationScriptSyntax.ts:142]: what about ticks? @aviv86 do we want to rename it and use less c# biased value?

File: src/Raven.Studio/typescript/viewmodels/database/tasks/transformationScriptSyntax.ts [hunk 6]
```diff
@@ -113,26 +126,55 @@ orderData.TotalCost = Math.round(orderData.TotalCost * 100) / 100;
...
+    
+    olapEtlSamplePartitionHtml = transformationScriptSyntax.highlightJavascript(transformationScriptSyntax.olapEtlSamplePartitionText);
 
-    olapEtlSampleHtml = ko.pureComputed(() => {
-        return Prism.highlight(transformationScriptSyntax.olapEtlSampleText, (Prism.languages as any).javascript);
-    });
+    static readonly olapEtlSampleNoPartitionText =
+`loadToOrders(noPartition(), {
+    // Data will Not be partitioned
+    Company : this.Company
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/database/tasks/transformationScriptSyntax.ts]: remove space before :

File: src/Raven.Studio/typescript/viewmodels/database/tasks/transformationScriptSyntax.ts [hunk 7]
```diff
@@ -113,26 +126,55 @@ orderData.TotalCost = Math.round(orderData.TotalCost * 100) / 100;
...
+});`;
     
+    olapEtlSampleNoPartitionHtml = transformationScriptSyntax.highlightJavascript(transformationScriptSyntax.olapEtlSampleNoPartitionText);
+
+    static readonly olapEtlSampleKeyText = 
+`var key = new Date(this.OrderedAt);
+loadToOrders(partitionBy(key), {
+    // The partition that will be created will be: "_partition={key}"
+    Company : this.Company
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/database/tasks/transformationScriptSyntax.ts]: Company: this.Company

File: src/Raven.Studio/typescript/viewmodels/database/tasks/transformationScriptSyntax.ts [hunk 8]
```diff
@@ -113,26 +126,55 @@ orderData.TotalCost = Math.round(orderData.TotalCost * 100) / 100;
...
 `var orderDate = new Date(this.OrderedAt);
 var year = orderDate.getFullYear();
 var month = orderDate.getMonth() + 1;
 
-// The order of the following array values determines the partitions order in parquet file path
-loadToOrders(partitionBy([['year', year], ['month', month], ['source', $custom_field]]),
-{
+loadToOrders(partitionBy(['year', year], ['month', month]), {
+    // The order of params in the partitionBy method determines the parquet file path
     Company : this.Company,
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/database/tasks/transformationScriptSyntax.ts]: same here

File: src/Raven.Studio/wwwroot/App/views/database/tasks/editOlapEtlTask.html [hunk 9]
```diff
@@ -219,6 +169,55 @@ <h3 class="margin-top margin-top-sm">No Olap Tables have been defined</h3>
...
+                                <button type="button" class="btn btn-default btn-sm" title="Cancel" data-bind="click: $root.cancelEditedOlapTable"><i class="icon-cancel"></i></button>
+                            </div>
+                        </div>
+                    </form>
+                </div>
+                <!-- The olap tables list -->
+                <div class="text-center text-muted" data-bind="if: editedOlapEtl().olapTables().length === 0 && !showEditOlapTableArea()">
+                    <i class="icon-lg icon-empty-set"></i>
+                    <h3 class="margin-top margin-top-sm">No settings have been defined.</h3>
+                    <h3 class="margin-top margin-top-sm">The default ID column name is <code>_id</code>. Click Add Settings to override this name per table.</h3>
```
Comment (ml054) [inline src/Raven.Studio/wwwroot/App/views/database/tasks/editOlapEtlTask.html:200]: Click 'Add Settings' to override this name per table.

File: src/Raven.Studio/typescript/viewmodels/database/tasks/transformationScriptSyntax.ts (hunk 8 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/database/tasks/transformationScriptSyntax.ts]: done

File: src/Raven.Studio/typescript/viewmodels/database/tasks/transformationScriptSyntax.ts (hunk 6 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/database/tasks/transformationScriptSyntax.ts]: done

File: src/Raven.Studio/typescript/viewmodels/database/tasks/transformationScriptSyntax.ts (hunk 7 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/database/tasks/transformationScriptSyntax.ts]: done

File: src/Raven.Studio/typescript/viewmodels/database/tasks/transformationScriptSyntax.ts (hunk 5 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/database/tasks/transformationScriptSyntax.ts:142]: @ppekrol @ayende  ?

**Review by ml054 — APPROVED** [review]

**Pull request merged**